client/src/components/molecules/CourtAppearanceList/index.tsx

```diff
@@ -1,35 +1,46 @@
-import { useState } from 'react'
+import { useEffect } from 'react'
 import ReactPaginate from 'react-paginate'
 
 import CourtAppearanceAccordion from '../CourtAppearanceAccordion'
-import { ICourtAppearance } from '~/shared/interfaces'
+import { useAppDispatch, useAppSelector } from '~/hooks/reduxSelector'
+import { fetchClientCourtAppearances } from '~/redux/court-appearance/courtAppearanceSlice'
+import { useRouter } from 'next/router'
 
-const CourtAppearancesList = ({
-  courtAppearances
-}: {
-  courtAppearances: ICourtAppearance[]
-}): JSX.Element => {
-  const [pageNumber, setPageNumber] = useState<number>(0)
+export interface ICourtAppearancesPayload {
+  clientId: number
+  page?: number
+}
+
+const CourtAppearancesList = (): JSX.Element => {
+  const dispatch = useAppDispatch()
+  const { query } = useRouter()
 
-  const listPerPage = 5
-  const pagesVisited = pageNumber * listPerPage
+  const { courtAppearances } = useAppSelector((state) => state.courtAppearance) || {}
 
-  const displayCourtAppearances = courtAppearances?.slice(pagesVisited, pagesVisited + listPerPage)
+  const getCourtAppearances = async (payload: ICourtAppearancesPayload) => {
+    await dispatch(fetchClientCourtAppearances(payload))
+  }
+  useEffect(() => {
+    getCourtAppearances({ clientId: Number(query?.id) })
+  }, [query])
 
-  const pageCount = Math.ceil(courtAppearances?.length / listPerPage)
+  const courtAppearancesList = courtAppearances?.data
+  const pageCount = courtAppearances?.meta?.last_page
 
-  const handlePageChange = ({ selected }: { selected: number }): void => setPageNumber(selected)
+  const handlePageChange = async ({ selected }: { selected: number }) => {
+    await dispatch(fetchClientCourtAppearances({ clientId: Number(query?.id), page: selected + 1 }))
+  }
 
   return (
     <>
-      {displayCourtAppearances?.map((courtAppearance) => {
+      {courtAppearancesList?.map((courtAppearance: any) => {
```

Comment:
If you have a proper type for courtAppearancesList you don't to add the :any during iteration but then I will prove this one

Reply:
@jasonclchua Just remove `any` from this one.